src/main/java/org/apache/sysml/hops/ipa/FunctionCallGraph.java

```diff
@@ -364,9 +369,9 @@ else if (sb instanceof FunctionStatementBlock) {
 						&& ((ExternalFunctionStatement)fsb.getStatement(0)).isSecondOrder() ) {
 						ret = true;
 					}
-				}
-				else if( HopRewriteUtils.isData(h, DataOpTypes.TRANSIENTWRITE)
-					&& HopRewriteUtils.isNary(h.getInput().get(0), OpOpN.EVAL) ) {
+				} else if (HopRewriteUtils.isData(h, DataOpTypes.TRANSIENTWRITE)
+						&& (HopRewriteUtils.isNary(h.getInput().get(0), OpOpN.EVAL)
+						|| HopRewriteUtils.isParameterBuiltinOp(h.getInput().get(0), Hop.ParamBuiltinOp.PARAMSERV))) {
```

Comment:
Since ParameterizedBuiltinOps can appear anyway in the DAG (not just under the root node) we need a to a DAG traversal to ensure that we catch it in the general case. Please move the probe outside the for loop though to ensure each DAG is traversed just once here.